apache-kafka-2/src/main/java/com/baeldung/kafka/message/ordering/payload/Message.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.kafka.message.ordering.payload;
+
+import java.util.Random;
+
+public class Message implements Comparable<Message> {
+    private long insertPosition;
+    private long messageId;
+
+    public Message(){
+
+    }
+
+    public Message(long insertPosition, long messageId) {
+        this.insertPosition = insertPosition;
+        this.messageId = messageId;
+    }
+
+    public long getInsertPosition() {
+        return insertPosition;
+    }
+
+    public long getMessageId() {
+        return messageId;
+    }
+
+    @Override
+    public int compareTo(Message other) {
+        return Long.compare(this.messageId, other.messageId);
+    }
+
+    public static long getRandomMessageId() {
```

Comment:
If the external message id is supposed to be used for ordering, why are we generating a random one? Also, I do not see any code that uses the message id to impose an external ordering.

Am I missing something here?

Can you also clarify what the insert position is supposed to represent? As far as I can see, it is used to form the key of the ProducerRecord, which means it is actually more like the "partition key" than an insert position.

Reply:
I have renamed insertPosition attribute to partitionKey and changed its data type as well. Idea behind messageId was that it would act as business entity identifier. I can renamed same to entityId or businessEntityIdentifier.
HTH